python/example/plasticity.py

```diff
@@ -0,0 +1,62 @@
+import arbor as A
+
+
+class recipe(A.recipe):
+    def __init__(self, n):
+        super().__init__()
+        self.ccp = A.neuron_cable_properties()
+        self.connected = {}
```

Comment:
Upon re-reading I got confused again: _both_ connections are added through `.add_connection()`.

For readability's sake I propose that we instantiate the recipe with an explicit connection dict.

```suggestion
        self.conn_list = {x:[] for x in range(nc)}
```

Reply:
> Upon re-reading I got confused again: both connections are added through .add_connection().

Yes, for homogeneity, the initial connection is just like others we might call into being.

> For readability's sake I propose that we instantiate the recipe with an explicit connection dict.

I am really not sure this is more readable. Also, it teaches bad habits: Doing it this way circumvents
the lazy construction we encourage. And for good reason, a recent experiment with the brunel.py example
that switched to an explicit connection list in Python blew up at ~4GB RAM used. The lazy version is fine,
though.

As written, we store the minimum data possible.

Reply:
The ingredient to `arbor.connection` are now spread out, so it's not immediately obvious what `.add_connection()` does. Alternatively, it can be renamed `.add_connection_to_cell_0()` or somesuch, to make it clear it creates very specific connections.

Reply:
Ok, that's verbose, but this _is_ an example after all.

Reply:
Done, the example also has way more comments now.